### ISSUE TITLE
Document how to update a git checkout of another branch

### ISSUE THREAD

**Issue by SpareSimian** [wxWidgets/wxWidgets#22695]
While I understand the idea behind git submodules, I'm not familiar with the commands to maintain them. I was able to switch my working directory from 3.1.2 to 3.2.0 with trial and error, updating the older libraries and initializing and pulling the new ones. It would be desirable to enhance the git-related documentation to list and explain the commands needed here.

**Comment by vadz** [comment]
Sorry, really not sure what would you wish to be documented here. wx doesn't do anything special here, i.e. we use the submodules in a fairly standard way, which means you just need to run `git submodule update` after switching branches, as usual.

I don't think we want to (incompletely and imperfectly) reproduce Git documentation, so I don't really see what could we improve here. If you do, please let us know, but otherwise I don't think it's useful to keep this open.

**Comment by SpareSimian** [comment]
I went down the wrong rabbit hole when I read this:

https://github.com/wxWidgets/wxWidgets/blob/master/docs/contributing/how-to-update-third-party-library.md

I thought I was supposed to go into each 3rdparty library and issue a new checkout to switch branches.  I did know that I could use "git submodule update --init" on a fresh checkout, as documented in README-GIT.md. So after an initial failure from using branch checkouts in the 3rdparty directories, I guessed that a simple submodule update from the root would work. So maybe add that explicitly for an additional step when one switches branches?

Normally I'd go to the git docs, and they're mostly pretty good. But the submodule page is pretty confusing to me. The explosion of subcommands and options leaves me wondering what magic is really going on here. Fortunately, the simple update subcommand was enough to get my build working again.
